### ISSUE TITLE
RLS: 1.2

### ISSUE THREAD

**Issue by simonjayhawkins** [pandas-dev/pandas#37784]
Tracking issue for the 1.2 release. https://github.com/pandas-dev/pandas/milestone/73

Planning a release candidate for sometime towards the end of the month, 1-2 weeks before the final release.

List of open regressions: https://github.com/pandas-dev/pandas/issues?q=is%3Aopen+is%3Aissue+label%3ARegression (includes regressions that may be included in 1.1.5)

List of open blockers: https://github.com/pandas-dev/pandas/issues?q=is%3Aopen+is%3Aissue+label%3Ablocker+

**Comment by simonjayhawkins** [comment]
nightly builds on master are currently failing. https://dev.azure.com/pandas-dev/pandas-wheels/_build/results?buildId=48433&view=results

the failures for 3.7 (for linux and macOS) are related to -Werror. (not applicable to Windows)

```
 #warning "Using deprecated NumPy API, disable it by " \
  ^
cc1: all warnings being treated as errors
```

These are not failing on 1.1.x. In setup.py on 1.1.x we have
```
# in numpy>=1.16.0, silence build warnings about deprecated API usage
#  we can't do anything about these warnings because they stem from
#  cython+numpy version mismatches.
macros.append(("NPY_NO_DEPRECATED_API", "0"))
if "-Werror" in extra_compile_args:
    try:
        import numpy as np
    except ImportError:
        pass
    else:
        if np.__version__ < LooseVersion("1.16.0"):
            extra_compile_args.remove("-Werror")
```
this has been removed on master in since the min numpy version for pandas 1.2 is now 1.16.5

in MacPython we have `NP_BUILD_DEP: "1.14.5"` for 3.7

we need to get the nightly builds to pass (and before 1.2rc0 release) while also allowing 1.1.5 to build.

in https://github.com/MacPython/pandas-wheels/pull/111 I have changed to `NP_BUILD_DEP: "1.16.5"` for 3.7 and indeed the py37 wheels are building.

in https://github.com/MacPython/pandas-wheels/pull/109 I have done the same to check that this change will not affect the 1.1.5 release

Will be opening a PR shortly on MacPython with this change. (if https://github.com/MacPython/pandas-wheels/pull/109 passes)

**Comment by simonjayhawkins** [comment]
the pip-test and conda-test stages of the release process are currently failing. https://github.com/simonjayhawkins/pandas-release/runs/1447063807?check_suite_focus=true

have reproduced this locally, so will investigate further.

**Comment by simonjayhawkins** [comment]
> the pip-test and conda-test stages of the release process are currently failing. https://github.com/simonjayhawkins/pandas-release/runs/1447063807?check_suite_focus=true

#35895 added the use of the `worker_id` fixture, which is only available if pytest-xdist is installed. see https://github.com/pandas-dev/pandas/pull/35895#discussion_r477018354

IIUC pytest-xdist is an optional dependency and pd.test is part of the public api https://pandas.pydata.org/docs/reference/api/pandas.test.html

cc @TomAugspurger

**Comment by jreback** [comment]
@pandas-dev/pandas-core don't tag any more issues /PRs for 1.2, unless they are actually ready to merge, better to just ping and say ready for 1.2

**Comment by simonjayhawkins** [comment]
wrt documentation

- there's a significant clean-up in #38002 to be merged before another pass.
- imo the tracebacks are a distraction https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.2.0.html#optionally-disallow-duplicate-labels - any alternative to this?
- need to add contributors
- if 1.2.0rc0 is released before 1.1.5 we may want to start the 1.2.x branch before tagging 1.2.0rc0 and remove the 1.1.5 release notes first
- have done dummy release locally for 1.2 and built the docs. issue #37813 is not an release environment issue, function signatures are fine.

**Comment by jorisvandenbossche** [comment]
> imo the tracebacks are a distraction https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.2.0.html#optionally-disallow-duplicate-labels - any alternative to this?

Yes, I also noticed that and wanted to do something about it. I think the easiest alternative for now is a plain code-block with the output we want (nicer general solution would be https://github.com/pandas-dev/pandas/issues/10715)

**Comment by simonjayhawkins** [comment]
1.1.5 is in a good state for release. Not sure whether the preference is to do 1.1.5 before/after 1.2.0rc0

I'll open an issue for tracking the 1.1.5 release shortly. I'm thinking, looking at the open issues, that will just remove milestones from all the open issues, with a message about a imminent release and stating community contributions welcome.

Some of the issues have now had the milestone changed several times, some from 1.0.x. So that's why I think we could now just remove the milestone altogether. If I do this today, that will give interested parties time to comment prior to the release.

Scheduled for Monday. but could do say 1.2.0rc0 Monday, 1.1.5 following Monday, and 1.2 the Monday after that.

thoughts?

**Comment by simonjayhawkins** [comment]
> Yes, I also noticed that and wanted to do something about it.

I think, unlike code changes, it's ok to tidy release notes between 1.2.0rc0 and 1.2, so not a blocker imo.

I'm going to look at the tag script later, so that the v1.2.0rc0 tag, the 1.2.x branch and v1.3.0.dev0 are created atomically, to try an deter any changes after tagging the release candidate other than reported bugs/regressions (with the rc). Any older bugs/regressions could perhaps wait for v1.3.1

**Comment by jorisvandenbossche** [comment]
I don't know if an rc for 1.2 on Monday is realistic, given the number of open issues? 

Regarding the issues tagged for 1.1.5: I agree that several of them can be removed from that milestone (eg some performance regressions of a year old that were already partly solved). But the actual behaviour regressions, if not removing from the 1.1.5 milestone should then be added to the 1.2 (or 1.2.1) milestone, IMO, so we don't loose track of them.

**Comment by jorisvandenbossche** [comment]
BTW, I have find it easier in the past to create a temporary milestone for "1.2.0rc", so we can distinguish between issues that only need to be fixed before the final 1.2 release, and issues that are blockers for the RC. After the RC we can put the issues labeled with 1.2.0rc back to the 1.2 milestone with a batch change, and remove the milestone. 
(just a suggestion, up to you)

**Comment by simonjayhawkins** [comment]
> BTW, I have find it easier in the past to create a temporary milestone for "1.2.0rc", so we can distinguish between issues that only need to be fixed before the final 1.2 release, and issues that are blockers for the RC. After the RC we can put the issues labeled with 1.2.0rc back to the 1.2 milestone with a batch change, and remove the milestone.
> (just a suggestion, up to you)

That sort of goes against my thinking that a release candidate is moreless final subject to any issues with the rc.

I'm not keen on doing a rc if there are more changes planned before the final release.

**Comment by simonjayhawkins** [comment]
> But the actual behaviour regressions, if not removing from the 1.1.5 milestone should then be added to the 1.2 (or 1.2.1) milestone, IMO, so we don't loose track of them.

on a similar vein to my above response. I wouldn't be keen on moving them to 1.2, ok with 1.2.1 though.

**Comment by jorisvandenbossche** [comment]
> That sort of goes against my thinking that a release candidate is moreless final subject to any issues with the rc.

Yep, I am certainly fine with that. In the past we have been more liberal with changes during the RC, but I think it would be good to change that to what you propose to be more strict. In which case another label might not be needed (there could be still issues only for the final release, but it should be only very few issues).

> I wouldn't be keen on moving them to 1.2, ok with 1.2.1 though.

Unless our goal is to actually fix them. IMO we should take the time to go through the remaining issues, identify the important regressions, and actually try to fix them before the release. Even if that delays the release a little bit, IMO that's still a good outcome. We have some support for maintenance now, and fixing release blockers can be part of that IMO.

**Comment by simonjayhawkins** [comment]
> I don't know if an rc for 1.2 on Monday is realistic, given the number of open issues?

only 3 are labelled as blockers. once/if https://github.com/MacPython/pandas-wheels/pull/113 is merged, the blocker label can be removed from #36429

if a non-blocking issue isn't fixed before the rc, then it'll have to wait for 1.2.1 or 1.3. (maybe sounds a bit harsh)

IIUC our release schedule is more calender based than feature based.

**Comment by jorisvandenbossche** [comment]
> IIUC our release schedule is more calender based than feature based.

I was not really talking about features, but about regressions, though

**Comment by jreback** [comment]
we need to finish off the PRs (about 10) & issues and either move them off / merge. i think by monday is doable.

**Comment by simonjayhawkins** [comment]
> > the pip-test and conda-test stages of the release process are currently failing. https://github.com/simonjayhawkins/pandas-release/runs/1447063807?check_suite_focus=true
> 
> #35895 added the use of the `worker_id` fixture, which is only available if pytest-xdist is installed. see [#35895 (comment)](https://github.com/pandas-dev/pandas/pull/35895#discussion_r477018354)
> 
> IIUC pytest-xdist is an optional dependency and pd.test is part of the public api https://pandas.pydata.org/docs/reference/api/pandas.test.html
> 
> cc @TomAugspurger

for now, if this is not blocking will add pytest-xdist to pip test and conda recipe to pandas-release, to get passing tests. tested at https://github.com/simonjayhawkins/pandas-release/actions/runs/388724204

**Comment by simonjayhawkins** [comment]
> we need to finish off the PRs (about 10) & issues and either move them off / merge. i think by monday is doable.

all the nightly wheels built on azure overnight. https://dev.azure.com/pandas-dev/pandas-wheels/_build/results?buildId=48896&view=results and https://anaconda.org/scipy-wheels-nightly/pandas/files

(no nightly wheels for arm, https://github.com/MacPython/pandas-wheels/issues/102)

so from the wheel building perspective, 1.2.0rc0 release is doable.

**Comment by jreback** [comment]
the only remaining PRs which we should do *before* the RC are:

- xlrd deprecation: https://github.com/pandas-dev/pandas/pull/35029
- ndim deprecation: https://github.com/pandas-dev/pandas/pull/38134

others can be for during the RC period is ok

**Comment by simonjayhawkins** [comment]
> the only remaining PRs which we should do _before_ the RC are:

I've added the blocker tag to these.

@jorisvandenbossche you can add the blocker tag to the PRs you've created today, if they need to be done before rc.

**Comment by jorisvandenbossche** [comment]
I would like to get in the fixes I am doing for FloatingArray: https://github.com/pandas-dev/pandas/pull/38178 (so that PR would need to get a review). 

For the rest, they are not a blocker for the RC, but still for 1.2, IMO. So if you are fine with doing some more fixes after the RC, that's fine.

BTW, it might make more sense to first do a 1.1.5 release, and afterwards cut the 1.2.0.RC. Because any fix going into 1.1.5, also needs to be in 1.1.2 (I think it wouldn't make sense to fix a regression in 1.1.5, but have it fail again on 1.2.0. But again, that depends on how liberal we want to be with more fixes after the RC.

**Comment by jreback** [comment]
I don't think it makes a big diff whether 1.1.5 or 1.20rc comes out first.

**Comment by simonjayhawkins** [comment]
> BTW, it might make more sense to first do a 1.1.5 release, and afterwards cut the 1.2.0.RC. Because any fix going into 1.1.5, also needs to be in 1.2.0 (I think it wouldn't make sense to fix a regression in 1.1.5, but have it fail again on 1.2.0. But again, that depends on how liberal we want to be with more fixes after the RC.

in https://github.com/pandas-dev/pandas/issues/37784#issuecomment-733715061, i suggested

> Scheduled for Monday. but could do say 1.2.0rc0 Monday, 1.1.5 following Monday, and 1.2 the Monday after that.

> I don't think it makes a big diff whether 1.1.5 or 1.20rc comes out first.

It just means two backports, but since OK to do regression fixes between 1.2.0rc0 and 1.2, gives another week to look into the outstanding regressions from 1.0.5

**Comment by jorisvandenbossche** [comment]
If we are doing more fixes after 1.2RC, then that's certainly fine for me as well (
just wanted to make sure we are on the same page, as before you noted you proposed to be more strict about what gets included after 1.2RC by directly branching. But of course, for regressions, even if not found by the RC itself but known already now, that of course makes sense to still add those to 1.2 after the RC anyway)

**Comment by simonjayhawkins** [comment]
> If we are doing more fixes after 1.2RC, then that's certainly fine for me as well (

regressions (on master and from 1.0.5) and doc cleanups ok imo

> just wanted to make sure we are on the same page, as before you noted you proposed to be more strict about what gets included after 1.2RC by directly branching. But of course, for regressions, even if not found by the RC itself but known already now, that of course makes sense to still add those to 1.2 after the RC anyway)

yes, we should branch (1.2.x) immediately after rc and will need to backport anything in 1.2 to 1.2.x and anything for 1.1.5 to 1.1.x and 1.2.x

**Comment by simonjayhawkins** [comment]
as we discussed (offline), since the completion of ArrowStringArray work has missed 1.2, delaying that work by a couple of more weeks is not an issue.

so my intention is after the 1.2.0rc release is to spend a week looking at the regressions from 1.0.5 (and outstanding backports, e.g. #37633). Then after 1.1.5, spend a week looking at regressions on master, e.g. #35804 before the 1.2 release.

i.e. 2 weeks dedicated to tackling the issues tagged as regressions https://github.com/pandas-dev/pandas/issues?q=is%3Aopen+is%3Aissue+label%3ARegression

**Comment by jorisvandenbossche** [comment]
@simonjayhawkins sounds good!

**Comment by simonjayhawkins** [comment]
so it would be good to get #38057 and #38099 into master (even if the backports are still in progress) before tagging 1.2.0rc0, but not necessarily blockers.

**Comment by simonjayhawkins** [comment]
@jorisvandenbossche did do want to revert #38148 before the rc or should we just move the release note for now.

**Comment by jbrockmendel** [comment]
Let's revert #38148 and get a non-copying version in place for the next release.

**Comment by simonjayhawkins** [comment]
> the only remaining PRs which we should do _before_ the RC are:
> 
> * xlrd deprecation: #35029
> * ndim deprecation: #38134
> 
> others can be for during the RC period is ok

It looks like xlrd deprecation is close to being ready and is currently the only blocker.

so we'll be ready to do the rc tomorrow.

add the blocker tag to any other PRs/issues as needed.

**Comment by jreback** [comment]
@simonjayhawkins i think go for release here. we can address the few remaining issues during the rc.

**Comment by jbrockmendel** [comment]
for the rest of us, does this mean we're in a (partial?) merge moratorium for a while?

**Comment by jreback** [comment]
no

simon is going to branch 1.2.x (i think he push the tag to the branch though maybe to master as well)

we can continue to merge for 1.2 but need to tag them as they need to be backport to the 1.2.x branch 

else's they will be on 1.3

**Comment by simonjayhawkins** [comment]
```
(pandas-release) simon@DESKTOP-4UB01H7:~/pandas-release/pandas (master) $ git show $(git merge-base 1.0.x master)
commit d3f08566a80239a18a813ebda9a2ebb0368b1dc5 (tag: v1.0.0rc0)
Author: Pandas Development Team <pandas-dev@python.org>
Date:   Thu Jan 9 15:24:50 2020 -0600

    RLS: 1.0.0rc0
(pandas-release) simon@DESKTOP-4UB01H7:~/pandas-release/pandas (master) $ git show $(git merge-base 1.1.x master)
commit d9fff2792bf16178d4e450fe7384244e50635733 (tag: v1.1.0)
Author: Pandas Development Team <pandas-dev@python.org>
Date:   Tue Jul 28 08:14:59 2020 -0500

    RLS: 1.1.0
(pandas-release) simon@DESKTOP-4UB01H7:~/pandas-release/pandas (master) $
```

for the 1.0.0rc0 we branched and tagged v1.1.0.dev0 straight away.

```
commit d3f08566a80239a18a813ebda9a2ebb0368b1dc5 (tag: v1.0.0rc0)
Author: Pandas Development Team <pandas-dev@python.org>
Date:   Thu Jan 9 15:24:50 2020 -0600

    RLS: 1.0.0rc0

commit b7fcb545a1a298817cd7d9f8940f19992d1202d2 (tag: v1.1.0.dev0)
Author: jbrockmendel <jbrockmendel@gmail.com>
Date:   Thu Jan 9 18:11:30 2020 -0800

    CLN: remove unnecessary overriding in subclasses (#30875)
```

whereas for 1.1.0rc0, we didn't branch till after the actual release (and had a short rc period)
```
(pandas-release) simon@DESKTOP-4UB01H7:~/pandas-release/pandas (master) $ git log --reverse v1.1.0rc0^..
commit bfac13628394ac7317bb94833319bd6bf87603af (tag: v1.1.0rc0)
Author: Pandas Development Team <pandas-dev@python.org>
Date:   Fri Jul 17 14:03:50 2020 -0500

    RLS: 1.1.0rc0

commit e673b6907c1ce6ef67d67dc3638310567d1c6415
Author: Tom Augspurger <TomAugspurger@users.noreply.github.com>
Date:   Mon Jul 20 17:17:50 2020 -0500

    CI: pin matplotlib for doc build (#35358)

commit 3e88e170a436b5e3cfdfc29f8a7416220658c616
Author: Simon Hawkins <simonjayhawkins@gmail.com>
Date:   Tue Jul 21 12:20:00 2020 +0100

    REGR: MultiIndex Indexing (#35353)

    * REGR: MultiIndex Indexing

    * add test

commit 06e2793a35e74303d7cc5dd47eb5959387244655
Author: Matthew Roeschke <emailformattr@gmail.com>
Date:   Wed Jul 22 03:45:40 2020 -0700

    Change defaults for rolling/expanding.apply engine kwargs to None (#35374)

    * Change defaults for rolling/expanding.apply engine kwargs to None

    * Add whatsnew

    Co-authored-by: Matt Roeschke <mroeschke@housecanary.com>

commit d3343d9e7f2aa93572d73335a1cc489db267d1ee
Author: Tom Augspurger <TomAugspurger@users.noreply.github.com>
Date:   Thu Jul 23 14:10:33 2020 -0500

    PKG: Set max pin for Cython (#35396)

    We know that pandas doesn't work with Cython 3.0
    (https://github.com/pandas-dev/pandas/issues/34213,
    https://github.com/pandas-dev/pandas/issues/34014)

    This sets the maximum supported version of Cython in our pyproject.toml
    to ensure that pandas 1.1.0 can continue to be built from source without
    Cython pre-installed after Cython 3.0 is released.

commit 00ea10c158ac3d4087f6010f375084e9ddfcab99
Author: Tom Augspurger <TomAugspurger@users.noreply.github.com>
Date:   Thu Jul 23 17:06:03 2020 -0500

    Matplotlib 3.3 compatibility fixups (#35393)

commit 8f54d1401e9924627618c74db9d4671b72b7cb2d
Author: alm <alonme@users.noreply.github.com>
Date:   Fri Jul 24 15:46:54 2020 +0300

    Remove leftover partial-result code (#35397)

commit 04e9e0afd476b1b8bed930e47bf60ee20fa78f38
Author: Richard Shadrach <45562402+rhshadrach@users.noreply.github.com>
Date:   Fri Jul 24 09:06:26 2020 -0400

    DOC: Fixed formatting and errors in whatsnew v1.1.0 (#35398)

commit 6302f7b98ad24adda2d5a98fef3956f04f28039d
Author: Joris Van den Bossche <jorisvandenbossche@gmail.com>
Date:   Mon Jul 27 20:00:06 2020 +0200

    REGR: revert ExtensionBlock.set to be in-place (#35271)

    * REGR: revert ExtensionBlock.set to be in-place

    Co-authored-by: Simon Hawkins <simonjayhawkins@gmail.com>
    Co-authored-by: Tom Augspurger <tom.w.augspurger@gmail.com>

commit 7f3502d6ed9ec9f1e1a555185e357ce339a8f4bc
Author: Tom Augspurger <TomAugspurger@users.noreply.github.com>
Date:   Tue Jul 28 08:11:26 2020 -0500

    DOC: 1.1.0 release date (#35435)

    * DOC: 1.1.0 release date

commit d9fff2792bf16178d4e450fe7384244e50635733 (tag: v1.1.0)
Author: Pandas Development Team <pandas-dev@python.org>
Date:   Tue Jul 28 08:14:59 2020 -0500

    RLS: 1.1.0

commit d1b1faa3f68dad6163000784c34a66b8f6c227e1 (tag: v1.2.0.dev0)
Author: Tom Augspurger <tom.w.augspurger@gmail.com>
Date:   Tue Jul 28 14:28:37 2020 -0500

    Start 1.2.0
```

we had a discussion on branching for 1.1.0 in #34730. 

although no backports were necessary, it did mean a merge moratorium, which I initially accepted https://github.com/pandas-dev/pandas/issues/34730#issuecomment-660067723 but after a few days was more -1 , see https://github.com/pandas-dev/pandas/issues/34730#issuecomment-662376857

so in this release, we'll tag 1.3.0.dev0 straight after 1.2.0rc0 (either tag the next commit or push an empty commit)

and then can add the 1.3 whatsnew and carry on merging to master straight away.

(will also need to activate ci for the newly created 1.2.x branch moreless straight away too)

**Comment by simonjayhawkins** [comment]
> i think he push the tag to the branch though maybe to master as well

tagging on the branch instead may be worth considering in the future, it may have some advantages for an automated release process. discussion in #21050

we currently tag on master so the base of the branch has the release tag.

**Comment by simonjayhawkins** [comment]
> tagging on the branch instead may be worth considering in the future, it may have some advantages for an automated release process. discussion in #21050

since we only have 1 blocker, we could also do this for this release.

i.e. create the 1.2.x branch anytime now and can merge stuff for 1.3 to master. once the blocker is merged to master, backport it to 1.2.x and then created the rc from there.

**Comment by jreback** [comment]
yeah i think we should tag master with 1.2.0rc0 when you branch, IIRC this make it easy to track things, @TomAugspurger would know better.

**Comment by simonjayhawkins** [comment]
there are issue with Travis credits on MacPython. see https://github.com/MacPython/scipy-wheels/issues/106#issuecomment-735359681

we only recently started to use Travis again on MacPython for the arm builds. https://github.com/MacPython/pandas-wheels/pull/99

The last month's usage of Travis can be seen at https://travis-ci.com/github/MacPython?tab=insights

I have disabled the cron job for the nightly build as it was erroring, not building against master and not producing wheels. https://github.com/MacPython/pandas-wheels/issues/102

This appears to have made a difference to the daily usage.

I have also disabled the Travis for pull requests, as it appears that alot of usage was from some pre-release testing/cleaning I did last week.

Since the credit usage is critical (https://github.com/MacPython/scipy-wheels/issues/106#issuecomment-737542345), I don't think we should build wheels for arm for the rc.

**Comment by jreback** [comment]
@simonjayhawkins i would simply release now

**Comment by simonjayhawkins** [comment]
we still have the 1 blocker. do the release candidate tomorrow regardless of whether #38140 is resolved? This effectively means that it should be deferred to 1.3 and not included between 1.2.0rc0 and 1.2.0

**Comment by jreback** [comment]
yes we can always push that issue if needbe or just merge it - but nothing is a blocker

**Comment by simonjayhawkins** [comment]
rc pre-release checklist

- [ ]  MacPython.pandas nightly wheels built ok https://dev.azure.com/pandas-dev/pandas-wheels/_build/results?buildId=49620&view=results
- [ ]  release readiness passing https://github.com/simonjayhawkins/pandas-release/actions/runs/407820772
- [ ]  ci checks https://github.com/pandas-dev/pandas/commits/master
- [ ]  no open blockers https://github.com/pandas-dev/pandas/milestone/73

**Comment by simonjayhawkins** [comment]
will start the release after the ci on the latest commit to master completes.

**Comment by jorisvandenbossche** [comment]
@simonjayhawkins if that's OK for you, I would propose to merge the last two remaining PRs that are tagged for 1.2 (it's a deprecation, so would be good to include in the RC). Taking a look now, but I think both are mergeable. Will only take a bit longer to have CI completing on master.

**Comment by simonjayhawkins** [comment]
no problem.

also can you merge the conda-forge PR for 1.1.5. (sometime before 1.2.0rc0 gets an automated PR generated)

**Comment by jorisvandenbossche** [comment]
@simonjayhawkins will do (can you comment on that PR (https://github.com/conda-forge/pandas-feedstock/pull/100) whether you want to be added as recipe-maintainer or not?)

**Comment by jorisvandenbossche** [comment]
OK, last two PRs are merged here for 1.2, so again waiting on CI to complete on master

**Comment by simonjayhawkins** [comment]
known ci failure
```
=========================== short test summary info ============================
FAILED pandas/tests/io/test_html.py::TestReadHtml::test_banklist_url_positional_match[bs4]
= 1 failed, 145976 passed, 3162 skipped, 1094 xfailed, 64 warnings in 2863.76s (0:47:43) =
```

starting release now.

- [ ] tag locally
- [ ] build sdist
- [ ] pip test
- [ ] conda test
- [ ] build docs
- [ ] push tag https://github.com/pandas-dev/pandas/releases/tag/v1.2.0rc0
- [ ] start new tag v1.3.0.dev0 https://github.com/pandas-dev/pandas/releases/tag/v1.3.0.dev0
- [ ] create 1.3 whatsnew
- [ ] create 1.2.x branch https://github.com/pandas-dev/pandas/tree/1.2.x
- [ ] activate ci on 1.2.x
- [ ] PR to build wheels https://github.com/MacPython/pandas-wheels/pull/111
- [ ] upload docs https://pandas.pydata.org/pandas-docs/version/1.2.0/
- [ ] wheels in staging area https://anaconda.org/multibuild-wheels-staging/pandas/files
- [ ] github release
- [ ] upload wheels to PyPI
- [N/A] change link for stable docs
- [ ] conda-forge bot picked up on release https://github.com/conda-forge/pandas-feedstock/pulls
- [ ] available on conda-forge

**Comment by jreback** [comment]
@simonjayhawkins branch already created i see, so we can resume merging for 1.3 and if anything needed for 1.2 can then backport to the branch, correct?

**Comment by simonjayhawkins** [comment]
that's correct.

**Comment by jorisvandenbossche** [comment]
> conda-forge bot picked up on release

@simonjayhawkins I don't think that the conda-forge bots work for RCs, so I think you will need to do this one manually, see eg Tom's PR for 1.1RC: https://github.com/conda-forge/pandas-feedstock/pull/84

**Comment by simonjayhawkins** [comment]
thanks for the heads up. conda-forge needs the sdist from the release assets, so can't do this till later in the process.

**Comment by simonjayhawkins** [comment]
> @simonjayhawkins I don't think that the conda-forge bots work for RCs, so I think you will need to do this one manually, see eg Tom's PR for 1.1RC: [conda-forge/pandas-feedstock#84](https://github.com/conda-forge/pandas-feedstock/pull/84) (and it's in a different branch)

@jorisvandenbossche not sure what else needs to change on the dev branch. for instance where are the 3.6 builds picked up from? The files with the configurations appear to be automatically generated files.

**Comment by jreback** [comment]
we need to release 1.2, so except for merging https://github.com/pandas-dev/pandas/pull/38571 (which is passing and lgtm). pls have a look @jorisvandenbossche @simonjayhawkins

**Comment by simonjayhawkins** [comment]
not had a response from @mroeschke on #38523. which should maybe be fixed for 1.2

**Comment by simonjayhawkins** [comment]
wheels are building on nightly. will open PR to test 1.2.x shortly after recent backports merged to 12.x

**Comment by simonjayhawkins** [comment]
have spent some time on https://github.com/pandas-dev/pandas/issues/38451, unsucessfully, but I see that has been moved off 1.2 (not a blokcer, will publish pdf docs with known issues)

**Comment by jorisvandenbossche** [comment]
It has been only a good week. If we want that people test the RC, we need to give a bit more time IMO

**Comment by jorisvandenbossche** [comment]
BTW, I checked some upstream projects: statsmodels has one failure (https://github.com/statsmodels/statsmodels/pull/7215), but didn't check in detail if its caused by a regression or was a bug in statsmodels. Seaborn is passing (https://github.com/mwaskom/seaborn/pull/2392). For dask, I opened a PR to test the RC (https://github.com/dask/dask/pull/6996) which has a bunch of failures, but didn't investigate yet.

**Comment by simonjayhawkins** [comment]
The most import outstanding regressions, all seem to be problematic to resolve

1. indexing DatetimeIndex with date object #35830 and #35466 (old behaviour should probably be restored and deprecated)
2. behavior change on DataFrame.apply() #35518 and #35517 (old behaviour should probably be restored)
3. different behaviour of df.isin() when df contains None (we are inconsistent in many places e.g. with df.eq)

**Comment by simonjayhawkins** [comment]
> It has been only a good week. If we want that people test the RC, we need to give a bit more time IMO

it's 2 weeks tomorrow, (or the next day for conda-forge wheels). The conda-forge wheel download numbers are disappointing... https://anaconda.org/conda-forge/pandas/files

**Comment by jreback** [comment]
sure we are going to have a 1.2.1 - let's not hold up anything (except for xlrd patch)

**Comment by jreback** [comment]
let's just release, we are planning a 1.2.1. people really do not test the rc0s so if we release we will have actual reports. waiting only begets waiting.

**Comment by simonjayhawkins** [comment]
final pre-release checklist

- [ ]  https://github.com/MacPython/pandas-wheels/pull/119
- [ ]  release readiness passing https://github.com/simonjayhawkins/pandas-release/actions?query=workflow%3A%22Tag+Release%22
- [ ]  release notes checked https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.2.0.html
- [ ]  ci checks https://github.com/pandas-dev/pandas/commits/1.2.x
- [ ]  no open issues/PRs (except this one) https://github.com/pandas-dev/pandas/milestone/73

**Comment by jorisvandenbossche** [comment]
Let's give it at least a few more days. As mentioned above, I started running the dask test suite, but the failures still need to be checked. I already identified one regression.

**Comment by jreback** [comment]
> Let's give it at least a few more days. As mentioned above, I started running the dask test suite, but the failures still need to be checked. I already identified one regression.

sure, but let's set a release date. how about wed/thurs (@simonjayhawkins according to your schedule). anything else can certainly wait till 1.2.1

**Comment by simonjayhawkins** [comment]
The date for me is not critical, since Christmas is cancelled. so anytime.

**Comment by mrocklin** [comment]
> The date for me is not critical, since Christmas is cancelled. so anytime.

Hi folks!  We were just discussing this on the Dask weekly call.  We're scrambling to address the breakages that this introduces in Dask, but are somewhat short-staffed this week due to the holidays.  If it's possible to delay releasing a bit while we get this sorted that would be welcome.

More broadly, this may be the first joint pandas+dask release since Tom left Anaconda and moved to Microsoft.  He historically smoothed over co-releases, and we need to build a process to fill that gap on our end going forward.

**Comment by jreback** [comment]
@mrocklin what exactly is failing in dask? I think @jorisvandenbossche pointed to a pandas regression, but AFAIK dask is not failing on anything? (or better to cross link.

happy to try to accomodate, but we are already way behind on this release. the .0 are typically not heavily tested anyhow. so delaying them doesn't actually do anything.

**Comment by mrocklin** [comment]
It looks like @jorisvandenbossche and @jsignell are working on it here: https://github.com/dask/dask/pull/6996

**Comment by simonjayhawkins** [comment]
> It looks like @jorisvandenbossche and @jsignell are working on it here: [dask/dask#6996](https://github.com/dask/dask/pull/6996)

@jorisvandenbossche is it just #38649 that would need to be included in 1.2 to cover the dask failures?

**Comment by jorisvandenbossche** [comment]
Yes, there are some more failures as well, but from a quick look those are either things that are known and need to updated in dask or are already fixed in pandas. So #38649 is the main one that would be good to have

**Comment by simonjayhawkins** [comment]
once #38678 is backported, will be ready for final pre-release checks

**Comment by simonjayhawkins** [comment]
final pre-release checklist

- [ ]  https://github.com/MacPython/pandas-wheels/pull/119
- [ ]  release readiness passing https://github.com/simonjayhawkins/pandas-release/actions?query=workflow%3A%22Tag+Release%22
- [ ]  release notes checked https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.2.0.html
- [ ]  ci checks https://github.com/pandas-dev/pandas/commits/1.2.x
- [ ]  no open issues/PRs (except this one) https://github.com/pandas-dev/pandas/milestone/73

**Comment by simonjayhawkins** [comment]
release notes checked - formatting issue in https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.2.0.html#consistency-of-dataframe-reductions

```In [26]: In [5]: df.all(bool_only=True)```

not a blocker

**Comment by simonjayhawkins** [comment]
ci checks

known failure
```
=========================== short test summary info ============================
FAILED pandas/tests/io/test_html.py::TestReadHtml::test_banklist_url_positional_match[bs4]
= 1 failed, 146190 passed, 3165 skipped, 1091 xfailed, 59 warnings in 2716.21s (0:45:16) =
```

not affecting wheel builds
```
=========================== short test summary info ============================
FAILED pandas/tests/arrays/categorical/test_warnings.py::TestCategoricalWarnings::test_tab_complete_warning
FAILED pandas/tests/frame/test_api.py::TestDataFrameMisc::test_tab_complete_warning[DataFrame]
FAILED pandas/tests/frame/test_api.py::TestDataFrameMisc::test_tab_complete_warning[Series]
FAILED pandas/tests/indexes/test_base.py::TestIndex::test_tab_complete_warning
FAILED pandas/tests/resample/test_resampler_grouper.py::test_tab_complete_ipython6_warning
= 5 failed, 145318 passed, 4090 skipped, 1090 xfailed, 6 warnings in 1050.35s (0:17:30) =
```

**Comment by simonjayhawkins** [comment]
starting release now.

- [ ] tag locally
- [ ] build sdist
- [ ] pip test
- [ ] conda test
- [ ] build docs
- [ ] push tag
- [ ] open PR to build wheels
- [ ] upload docs https://pandas.pydata.org/pandas-docs/version/1.2.0/
- [ ] wheels in staging area https://anaconda.org/multibuild-wheels-staging/pandas/files
- [ ] github release
- [ ] upload wheels to PyPI
- [ ] change link for stable docs
- [ ] conda-forge bot picked up on release https://github.com/conda-forge/pandas-feedstock/pulls
- [ ] available on conda-forge

**Comment by simonjayhawkins** [comment]
3.7 arm build failed. moving forward with release without it.

**Comment by jorisvandenbossche** [comment]
@simonjayhawkins thanks for the release!

You didn't yet send an announcement email?

(also, if you do, could you use the "1.2" link instead of "1.2.0"? Eg https://pandas.pydata.org/pandas-docs/version/1.2/whatsnew/v1.2.0.html instead of https://pandas.pydata.org/pandas-docs/version/1.2.0/whatsnew/v1.2.0.html)

**Comment by simonjayhawkins** [comment]
> 3.7 arm build failed. moving forward with release without it.

closing as discussion on official aarch support till open #33971
